public/apps/configuration/panels/audit-logging/audit-logging-general-settings.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPageHeader,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiGlobalToastList,
+} from '@elastic/eui';
+import { cloneDeep, set } from 'lodash';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import { AppDependencies } from '../../../types';
+import { EditSettingGroup } from './edit-setting-group';
+import { SETTING_GROUPS } from './constants';
+import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
+import { updateAuditLogging } from '../../utils/audit-logging-view-utils';
+
+export function AuditLoggingEditGeneralSetting(props: AppDependencies) {
+  const [config, setConfig] = useState<object>({});
+  const [editConfig, setEditConfig] = useState<object>({});
+
+  const [toasts, setToasts] = useState<Toast[]>([]);
+  const addToast = useCallback((toastToAdd: Toast) => {
+    setToasts((state) => state.concat(toastToAdd));
+  }, []);
+  const removeToast = (toastToDelete: Toast) => {
+    setToasts(toasts.filter((toast) => toast.id !== toastToDelete.id));
+  };
+
+  const handleChange = (setting, val) => {
+    const editedConfig = set(cloneDeep(editConfig), setting.path, val);
+
+    setEditConfig(editedConfig);
+  };
+
+  useEffect(() => {
+    const fetchConfig = async () => {
+      props.coreStart.http
+        .get(API_ENDPOINT_AUDITLOGGING)
+        .then((resp) => {
+          const responseConfig = resp.data.config;
+
+          setConfig(responseConfig);
+          setEditConfig(responseConfig);
+        })
+        .catch((e) => {
+          console.log(e);
+        });
```

Comment:
Suggest being consistent between async and promise. Since other written code are already written in async way, suggest change to async/await here.

Reply:
Sure